### PR TITLE
OpenSearch and GraphQL for multiple source filters

### PR DESCRIPTION
**Pull request by JPrevost**
Why are these changes being introduced:

* Being able to search multiple source at once is desired

Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/RDI-170 [FYI: This is just the OpenSearch/GraphQL portion of that ticket. The TIMDEX-UI portion should have been a separate ticket but currently isn't]

How does this address that need:

* Creates a method to add "OR" filtering in OpenSearch
* Updates GraphQL to expect source as an array of keywords instead of
  a single string

Document any side effects to this change:

* GraphQL previously expected Source as a single string and now expects
  an array of strings. As V2 is not yet launched, this isn't a concern
  for V2, but may require changes for V1 consumers including Bento which
  filters to just ArchivesSpace.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [ ] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Changes GraphQL Schema

YES, in a way that might cause issues when we enable v2 (haven't confirmed fully yet)

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

**Review by jazairi — CHANGES_REQUESTED** [review]
I think a comment needs to be updated in one of the tests, but otherwise this is looking great and seems to be working as expected. :shipit: 

Good call-out about how this might affect V1 consumers. I think we can add that to the checklist before we deploy to prod. Fwiw, I seem to be able to filter by source using the V1 endpoint on this PR, but I'd want to do a little more testing to confirm that.

File: test/controllers/graphql_controller_v2_test.rb [hunk 1]
```diff
@@ -314,4 +314,102 @@ def setup
     json = JSON.parse(response.body)
     assert(json['errors'].first['message'].present?)
   end
+
+  test 'graphqlv2 filter multiple sources' do
+    VCR.use_cassette('graphql v2 filter multiple sources') do
+
+      # no filters return 4 sources
```
[the remark this answers is not in the document]
Comment (JPrevost) [reply test/controllers/graphql_controller_v2_test.rb:321]: doh, left that comment in and refactored the test to not be based on a specific number and instead just "there should be more than the filtered count" to make this less fragile. I'll adjust or remove the comments.

File: test/controllers/graphql_controller_v2_test.rb [hunk 2]
```diff
@@ -314,4 +314,102 @@ def setup
...
+              }
+            }
+          }
+        }'
+      }
+
+      json = JSON.parse(response.body)
+      initial_source_array = json['data']['search']['aggregations']['source']
+
+      # filtering to 2 sources returns 2 sources
```
[the remark this answers is not in the document]
Comment (JPrevost) [reply test/controllers/graphql_controller_v2_test.rb:388]: Yup. 🙈

File: test/controllers/graphql_controller_v2_test.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jazairi) [reply test/controllers/graphql_controller_v2_test.rb:321]: Ah, that explains. Yeah, I agree that this testing approach is more robust.

**Pull request merged**